external-plugins/test-subset/plugin/main.go

```diff
@@ -0,0 +1,200 @@
+	fs.StringVar(&o.prowConfigPath,
+		"prow-config-path",
+		"",
+		"Path to Prow configuration (required).")
+	fs.StringVar(&o.jobsConfigBase,
+		"jobs-config-base",
+		"",
+		"Base path to a directory with Prow job configs (required).")
+	fs.StringVar(&o.jobsNs,
+		"jobs-namespace",
+		"",
+		"The namespace in which Prow jobs should be created.")
+	fs.StringVar(&o.cacheDir,
+		"cache-dir",
+		"",
+		"Directory to store git repos cache in.")
+	fs.StringVar(&o.prowLocation,
+		"prow-location",
+		"",
+		"Prow raw git location")
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		group.AddFlags(fs)
+	}
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func clientFactoryCacheDirOpt(cacheDir string) func(opts *gitv2.ClientFactoryOpts) {
+	return func(cfo *gitv2.ClientFactoryOpts) {
+		cfo.CacheDirBase = &cacheDir
+	}
+}
+
+func main() {
+	opts := gatherOptions()
+	opts.validate()
+
+	logger := setupLogger()
+	logger.Infoln("Setting up events server")
+
+	var config *rest.Config
+	var err error
+	if opts.kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", opts.kubeconfig)
+		mustSucceed(err, "Could not instantiate K8s config from the given kubeconfig.")
+	} else {
+		config, err = rest.InClusterConfig()
+		mustSucceed(err, "Could not instantiate K8s config from the in cluster config.")
+	}
+	prowClient, err := v1.NewForConfig(config)
+	mustSucceed(err, "Could not instantiate a Prow client from the given kubeconfig.")
+
+	if err := secret.Add(opts.github.TokenPath, opts.hmacSecretFile); err != nil {
+		logrus.WithError(err).Fatalf("Failed to start secrets agent.")
+	}
+
+	githubClient, err := opts.github.GitHubClient(opts.dryRun)
+	mustSucceed(err, "Could not instantiate github client.")
+
+	gitClientFactory, err := gitv2.NewClientFactory(clientFactoryCacheDirOpt(opts.cacheDir))
+	mustSucceed(err, "Could not instantiate git client factory")
+
+	eventsChan := make(chan *handler.GitHubEvent)
+
+	eventsHandler := handler.NewGitHubEventsHandler(
+		eventsChan,
+		logger,
+		prowClient.ProwJobs(opts.jobsNs),
+		githubClient,
+		opts.prowConfigPath,
+		opts.jobsConfigBase,
+		opts.prowLocation,
+		gitClientFactory)
+
+	eventsServer := server.NewGitHubEventsServer(secret.GetTokenGenerator(opts.hmacSecretFile), eventsHandler)
+
+	serverMux := http.NewServeMux()
+	serverMux.Handle(opts.endpoint, eventsServer)
+	srv := &http.Server{Addr: fmt.Sprintf(":%d", opts.port), Handler: serverMux}
+	interrupts.ListenAndServe(srv, 5*time.Second)
+	logger.Infoln("Events server is listening on port:", opts.port)
+	externalplugins.ServeExternalPluginHelp(serverMux, logger.WithField("plugin-help", ""), helpProvider)
+	interrupts.WaitForGracefulShutdown()
+	logger.Println("Test-subset plugin server was gracefully shut down")
+}
+
+func helpProvider(_ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The Test-subset plugin is used to trigger a job that run custom subset of the tests`,
```

Comment:
since the prow instance will display the help [here](https://prow.ci.kubevirt.io/plugins) and [here](https://prow.ci.kubevirt.io/command-help#test), and this one is similar to the `trigger` plugin we should structure the help content similar, i.e. at least tell users looking at the plugin docs how to use it and what command(s) are available.

See trigger plugin helpProvider as example here: https://github.com/kubernetes-sigs/prow/blob/a3626d1e12d61ca2c560a6a6ff12d310c101a319/pkg/plugins/trigger/trigger.go#L83

Note: you needn't aim to be as comprehensive but at least the bar minimum should be provided.

![image](https://github.com/user-attachments/assets/03822747-9b70-44ae-bde6-f30bb4606298)

Reply:
as example i guess you meant this link
https://github.com/kubernetes-sigs/prow/blob/a3626d1e12d61ca2c560a6a6ff12d310c101a319/pkg/plugins/trigger/trigger.go#L106

thanks